be/src/http/action/transaction_stream_load.cpp

```diff
@@ -144,15 +178,30 @@ void TransactionStreamLoadAction::_send_error_reply(HttpRequest* req, const Stat
     HttpChannel::send_reply(req, str);
 }
 
+void TransactionStreamLoadAction::_finish_and_reply(HttpRequest* req, const std::string& reply) {
+    ResourceHandler* handler = static_cast<ResourceHandler*>(req->handler_ctx());
+    if (handler != nullptr) {
+        // release StreamLoadContext lock before sending reply to the client,
+        // otherwise client may meet TXN_IN_PROCESSING error. The reason is that
+        // the client can send another load request quickly after receiving the
+        // reply, but the lock has not been released, so the new request can not
+        // acquire the lock, and meet the TXN_IN_PROCESSING error.
+        handler->release();
```

Comment:
is it necessary to free the StreamLoadContext itself together with the unlock, or unlock the lock is good enough?

Wondering if this ResourceHandler wrapper is absolutely necessary?

Reply:
1. unlock is enough to solve the problem, but the context is useless after the reply, so free the context at the same time
2. `ResourceHandler` records whether the resources have already been released to avoid duplicate releases in free_handler_ctx

Reply:
`free_handler_ctx` also need to know whether the unlock has been executed because `free_handler_ctx` will always be called for a normal and bad request